### PR TITLE
Point cloud 2d RGB renderer GUI

### PR DESCRIPTION
**Pull request by nyalldawson**
(Builds off https://github.com/qgis/QGIS/pull/40013)

This adds all the framework for creating widgets for configuring 2d renderers for point clouds, and implements a configuration widget for the RGB point cloud renderer:

Peek 2020-11-13 15-50

For GUI/functionality consistency I've based this widget heavily on the raster multiband renderer configuration, and have reused the QgsContrastEnhancement class here.

### CODE REVIEW

**Comment by Jean-Roc** [comment]
@nyalldawson Hi, I downloaded the mxe build to test this feature (and the whole point cloud support of course), the data source manager does not show a point cloud menu as seen on a few twitter captures and the *Add a point cloud layer* open the vector tile menu. Is it due to a missing dependancy in the build or a bug (I'll fill the issue then) ?

**Comment by nyalldawson** [comment]
@Jean-Roc it's a missing dependancy

**Comment by Jean-Roc** [comment]
Thanks for the answer Nyall, I'll wait for an available windows build to try it out

**Comment by nyalldawson** [comment]
@PeterPetrik this is ready for review now!

File: src/app/pointcloud/qgspointcloudlayerproperties.cpp [hunk 1]
```diff
@@ -110,12 +131,17 @@ void QgsPointCloudLayerProperties::apply()
...
   // TODO -- move to proper widget classes!
 
+#if 0
   std::unique_ptr< QgsDummyPointCloudRenderer > renderer = qgis::make_unique< QgsDummyPointCloudRenderer >();
   renderer->setAttribute( mAttributeComboBox->currentAttribute() );
   renderer->setZMin( mMinZSpin->value() );
   renderer->setZMax( mMaxZSpin->value() );
   renderer->setColorRamp( mBtnColorRamp->colorRamp() );
   mLayer->setRenderer( renderer.release() );
+#endif
```
Comment (PeterPetrik) [inline src/app/pointcloud/qgspointcloudlayerproperties.cpp:141]: remove?

File: src/gui/pointcloud/qgspointcloudrendererpropertieswidget.cpp [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+  // get rid of old active widget (if any)
+  if ( mActiveWidget )
+  {
+    stackedWidget->removeWidget( mActiveWidget );
+
+    delete mActiveWidget;
+    mActiveWidget = nullptr;
+  }
+
+  QgsPointCloudRendererWidget *w = nullptr;
```
Comment (PeterPetrik) [inline src/gui/pointcloud/qgspointcloudrendererpropertieswidget.cpp:180]: i know FORTRAN is popular, but i prefer readable variable names :)

**Review by PeterPetrik — APPROVED** [review]

File: src/app/pointcloud/qgspointcloudlayerproperties.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/app/pointcloud/qgspointcloudlayerproperties.cpp:141]: Can I leave it for now? I'll need to use this code (or similar) when adding a shade-by-attribute renderer....

File: src/gui/pointcloud/qgspointcloudrendererpropertieswidget.cpp (hunk 2 above)
Comment (nyalldawson) [reply src/gui/pointcloud/qgspointcloudrendererpropertieswidget.cpp:180]: Done

Comment (PeterPetrik) [reply src/gui/pointcloud/qgspointcloudrendererpropertieswidget.cpp]: sure np, that is why I selected "approved" as status of review :)

**Pull request merged**